lib/templates/template.tex

```diff
@@ -1,5 +1,8 @@
 \documentclass[$if(fontsize)$$fontsize$$else$12pt$endif$,$if(lang)$$lang$,$endif$$if(papersize)$$papersize$,$endif$$for(classoption)$$classoption$$sep$,$endfor$]{$documentclass$}
 
+\providecommand{\tightlist}{%
```

Comment:
Which raises the question of.. merging the pandoc default against our template, but at which version of pandoc, etc.  Tricky.

Not sure what to suggest about that. We could mandate a version.

```
$ pandoc --print-default-template latex > /tmp/pandoc-default
$ diff /tmp/pandoc-default underscore-ebook-template/lib/templates/template.tex
```

produces quite a few changes to review:

```diff
1c1,17
< \documentclass[$if(fontsize)$$fontsize$,$endif$$if(lang)$$lang$,$endif$$if(papersize)$$papersize$,$endif$$for(classoption)$$classoption$$sep$,$endfor$]{$documentclass$}
---
> \documentclass[$if(fontsize)$$fontsize$$else$12pt$endif$,$if(lang)$$lang$,$endif$$if(papersize)$$papersize$,$endif$$for(classoption)$$classoption$$sep$,$endfor$]{$documentclass$}
> 
> \providecommand{\tightlist}{%
>   \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
> 
> % Colors ----------------------------------------
> 
> \usepackage[xcolor]{mdframed}
> \definecolor{covercolor}{HTML}{$if(coverColor)$$coverColor$$else$777777$endif$}
> % \definecolor{primarycolor}{HTML}{428BCA}
> % \definecolor{successcolor}{HTML}{5CB85C}
> \definecolor{warningcolor}{HTML}{F0AD4E}
> \definecolor{dangercolor}{HTML}{D9534F}
> \definecolor{infocolor}{HTML}{5BC0DE}
> 
> % Fonts -----------------------------------------
> 
6a23
> 
30c47,54
<     \setmainfont{$mainfont$}
---
>     \setmainfont$mainfont$
> $else$
>     \setmainfont[%
>       Color=primary,
>       Path=$lib-dir$/fonts/Lato/,
>       BoldItalicFont=Lato-BlackItalic,
>       BoldFont=Lato-Bold,
>       ItalicFont=Lato-Italic]{Lato-Regular}
31a56
> 
33a59,65
> $else$
>     \setsansfont[%
>       Color=primary,
>       Path=$lib-dir$/fonts/Lato/,
>       BoldItalicFont=Lato-BlackItalic,
>       BoldFont=Lato-Bold,
>       ItalicFont=Lato-Italic]{Lato-Regular}
34a67
> 
36a70,71
> $else$
>     \setmonofont[Mapping=tex-ansi]{Menlo}
37a73
> 
41a78,80
> 
> % -----------------------------------------------
> 
46,47c85,86
< \usepackage{microtype}
< \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
---
>   \usepackage{microtype}
>   \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
52,77d90
< \ifxetex
<   \usepackage[setpagesize=false, % page size defined by xetex
<               unicode=false, % unicode breaks when used with xetex
<               xetex]{hyperref}
< \else
<   \usepackage[unicode=true]{hyperref}
< \fi
< \hypersetup{breaklinks=true,
<             bookmarks=true,
<             pdfauthor={$author-meta$},
<             pdftitle={$title-meta$},
<             colorlinks=true,
<             citecolor=$if(citecolor)$$citecolor$$else$blue$endif$,
<             urlcolor=$if(urlcolor)$$urlcolor$$else$blue$endif$,
<             linkcolor=$if(linkcolor)$$linkcolor$$else$magenta$endif$,
<             pdfborder={0 0 0}}
< \urlstyle{same}  % don't use monospace font for urls
< $if(lang)$
< \ifxetex
<   \usepackage{polyglossia}
<   \setmainlanguage{$mainlang$}
<   \setotherlanguages{$for(otherlang)$$otherlang$$sep$,$endfor$}
< \else
<   \usepackage[shorthands=off,$lang$]{babel}
< \fi
< $endif$
84,86c97,99
< $for(bibliography)$
< \addbibresource{$bibliography$}
< $endfor$
---
> $if(biblio-files)$
> \bibliography{$biblio-files$}
> $endif$
93a107,109
> 
> % Syntax highlighting ---------------------------
> 
97c113,115
< $if(verbatim-in-note)$
---
> 
> % -----------------------------------------------
> 
99,100d116
< \VerbatimFootnotes
< $endif$
105c121
< \usepackage{graphicx,grffile}
---
> \usepackage{graphicx}
114a131,152
> \ifxetex
>   \usepackage[setpagesize=false, % page size defined by xetex
>               unicode=false, % unicode breaks when used with xetex
>               xetex]{hyperref}
> \else
>   \usepackage[unicode=true]{hyperref}
> \fi
> \hypersetup{%
>   breaklinks=true,
>   bookmarks=true,
>   pdfauthor={$author-meta$},
>   pdftitle={$title-meta$},
>   colorlinks=true,
>   citecolor=$if(citecolor)$$citecolor$$else$blue$endif$,
>   urlcolor=$if(urlcolor)$$urlcolor$$else$blue$endif$,
>   linkcolor=$if(linkcolor)$$linkcolor$$else$magenta$endif$,
>   pdfborder={0 0 0}}
> 
> \urlstyle{same} % don't use monospace font for urls
> 
> % Links as footnotes ----------------------------
> 
127,128d164
< \providecommand{\tightlist}{%
<   \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
136a173,182
> $if(lang)$
> \ifxetex
>   \usepackage{polyglossia}
>   \setmainlanguage{$mainlang$}
> \else
>   \usepackage[$lang$]{babel}
> \fi
> $endif$
> 
> % Document metadata -----------------------------
149,157c195,241
< % Redefines (sub)paragraphs to behave more like sections
< \ifx\paragraph\undefined\else
< \let\oldparagraph\paragraph
< \renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
< \fi
< \ifx\subparagraph\undefined\else
< \let\oldsubparagraph\subparagraph
< \renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
< \fi
---
> % Utilities for the title page ------------------
> 
> \usepackage{tikz}
> 
> \newenvironment{bottompar}{\par\vspace*{\fill}}{\clearpage}
> 
> % Column support (see columns.coffee) -----------
> 
> \usepackage{multicol}
> 
> % Callout support (see callout.coffee) ----------
> 
> \newmdenv[%
>   linecolor=infocolor,linewidth=5pt,
>   topline=false,bottomline=false,rightline=false,
>   innertopmargin=1em,innerbottommargin=1em,
>   splittopskip=1em,splitbottomskip=1em,
>   backgroundcolor=infocolor!30,
>   frametitle=Tip]{InfoCallout}
> \newmdenv[%
>   linecolor=warningcolor,linewidth=5pt,
>   topline=false,bottomline=false,rightline=false,
>   innertopmargin=1em,innerbottommargin=1em,
>   splittopskip=1em,splitbottomskip=1em,
>   backgroundcolor=warningcolor!30,
>   frametitle=Advanced]{WarningCallout}
> \newmdenv[%
>   linecolor=dangercolor!60,linewidth=5pt,
>   topline=false,bottomline=false,rightline=false,
>   innertopmargin=1em,innerbottommargin=1em,
>   splittopskip=1em,splitbottomskip=1em,
>   backgroundcolor=dangercolor!25,
>   frametitle=Warning]{DangerCallout}
> 
> % \newmdenv[%
> %   topline=false,bottomline=false,leftline=false,rightline=false,
> %   innertopmargin=1em,innerbottommargin=1em,
> %   splittopskip=1em,splitbottomskip=1em,
> %   backgroundcolor=black!10]{Shaded}
> 
> % Customise default code blocks -----------------
> 
> \DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\},fontsize=\small}
> 
> % ===============================================
> % Document
> % ===============================================
159a244,266
> \pagecolor{white}
> 
> % Title page ------------------------------------
> 
> \begin{titlepage}
> \pagecolor{covercolor}
> \color{white}
> 
> \begin{tikzpicture}[overlay, remember picture]
> \node[anchor=north west, xshift=-.2cm, yshift=.2cm]
>      at (current page.north west)
>      {\includegraphics{$lib-dir$/templates/images/hero-left-overlay-white.pdf}};
> \end{tikzpicture}
> 
> \begin{tikzpicture}[overlay, remember picture]
> \node[anchor=south east, xshift=.2cm, yshift=-.2cm]
>      at (current page.south east)
>      {\includegraphics{$lib-dir$/templates/images/hero-right-overlay-white.pdf}};
> \end{tikzpicture}
> 
> \begin{center}
> \vspace*{5cm}
> 
161c268,286
< \maketitle
---
> {\fontsize{1.5cm}{1em} \textsf{$title$}}
> $endif$
> 
> $if(subtitle)$
> \vspace{.25em}
> 
> {\fontsize{1cm}{1em} \textsf{$subtitle$}}
> $endif$
> 
> \vspace{1.25em}
> 
> $for(author)$
> {\fontsize{1cm}{1em} \textsf{$author$}}
> $endfor$
> 
> \vspace{1em}
> 
> $if(date)$
> {\fontsize{.5cm}{1em} \textsf{$date$}}
162a288,309
> 
> \begin{bottompar}
> \begin{center}
> \includegraphics[width=5cm]{$lib-dir$/templates/images/brand.pdf}
> 
> \vspace{1cm}
> 
> Copyright $copyright$ $author$.
> \end{center}
> \end{bottompar}
> 
> \end{center}
> \end{titlepage}
> 
> % $if(title)$
> % \maketitle
> % $endif$
> 
> \pagecolor{white}
> 
> % Abstract --------------------------------------
> 
168a316,333
> % Include-before hook ---------------------------
> 
> \vspace*{\fill}
> 
> \begin{center}
> 
> {\Large $title$$if(subtitle)$ $subtitle$$endif$}
> 
> $date$
> 
> Copyright $copyright$ $author$.
> 
> Published by \href{http://underscore.io}{Underscore Consulting LLP}, Brighton, UK.
> 
> \end{center}
> 
> \vspace{1em}
> 
172a338,342
> 
> \vspace*{\fill}
> 
> % Tables of contents, etc. ----------------------
> 
175,176c345,346
< \hypersetup{linkcolor=$if(toccolor)$$toccolor$$else$black$endif$}
< \setcounter{tocdepth}{$toc-depth$}
---
> \hypersetup{linkcolor=black}
> \setcounter{tocdepth}{$if(toc-depth)$$toc-depth$$else$3$endif$}
185a356,358
> 
> % Main document body ----------------------------
> 
187a361,362
> % Bibliography ----------------------------------
> 
189c364
< $if(bibliography)$
---
> $if(biblio-files)$
197c372
< \bibliography{$for(bibliography)$$bibliography$$sep$,$endfor$}
---
> \bibliography{$biblio-files$}
204a380,382
> 
> % Include-after hook ----------------------------
> 
208a387,389
> 
> % The end ---------------------------------------
> 
```